samples/cpp/hello_reshape_ssd/main.cpp

```diff
@@ -2,214 +2,149 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 
-#include <format_reader_ptr.h>
-
-#include <inference_engine.hpp>
 #include <memory>
-#include <ngraph/ngraph.hpp>
-#include <samples/common.hpp>
+#include <sstream>
 #include <string>
 #include <vector>
 
+// clang-format off
+#include "ngraph/ngraph.hpp"
+#include "openvino/openvino.hpp"
+
+#include "samples/args_helper.hpp"
+#include "samples/common.hpp"
+#include "samples/slog.hpp"
+#include "format_reader_ptr.h"
 #include "reshape_ssd_extension.hpp"
+// clang-format on
 
-using namespace InferenceEngine;
+using namespace ov::preprocess;
 
 int main(int argc, char* argv[]) {
     try {
-        // ------------------------------ Parsing and validation of input arguments
-        // ---------------------------------
+        // -------- Get OpenVINO runtime version --------
+        slog::info << *ov::get_openvino_version() << slog::endl;
+
+        // --------------------------- Parsing and validation of input arguments
         if (argc != 5) {
             std::cout << "Usage : " << argv[0] << " <path_to_model> <path_to_image> <device> <batch>" << std::endl;
             return EXIT_FAILURE;
         }
-        const std::string input_model{argv[1]};
-        const std::string input_image_path{argv[2]};
+        const std::string model_path{argv[1]};
+        const std::string image_path{argv[2]};
         const std::string device_name{argv[3]};
         const size_t batch_size{std::stoul(argv[4])};
         // -----------------------------------------------------------------------------------------------------
 
-        // --------------------------- Step 1. Initialize inference engine core
-        // -------------------------------------
-        Core ie;
+        // Step 1. Initialize inference engine core
+        ov::runtime::Core core;
+        // -----------------------------------------------------------------------------------------------------
+
+        // Step 2. Read a model
+        slog::info << "Loading model files: " << model_path << slog::endl;
+        auto model = core.read_model(model_path);
+        printInputAndOutputsInfo(*model);
 
-        IExtensionPtr inPlaceExtension;
-        if (device_name.find("CPU") != std::string::npos) {
-            inPlaceExtension = std::make_shared<InPlaceExtension>();
-            // register sample's custom kernel (CustomReLU)
-            ie.AddExtension(inPlaceExtension);
```

Comment:
since you removed extensions adding, do we need `reshape_ssd_extension.hpp` then? 
This sample also demonstrates how to add extension w/o creating a shared library

Reply:
Probably, we can move this extension to nrgaph_function creation? In this case we can create a function with custom operation and infer it?